test/test_cuda.py

```diff
@@ -425,6 +425,167 @@ def tmp(self):
 
 
 class TestCuda(TestCase):
+
+    @staticmethod
+    def _test_memory_stats_generator(self, device=None, N=35):
+        if device is None:
+            device = torch.cuda.current_device()
+
+        m0 = torch.cuda.memory_allocated(device)
+        last_m_arr = [torch.cuda.memory_allocated(device)]
+        max_m_arr = [torch.cuda.max_memory_allocated(device)]
+        last_c_arr = [torch.cuda.memory_cached(device)]
+        max_c_arr = [torch.cuda.max_memory_cached(device)]
+
+        def alloc(*size):
+            with torch.cuda.device(device):
+                # NOTE: do **not** use methods that will have additional
+                #       overhead, e.g., inplace random sampling methods.
```

Comment:
why?

Reply:
First time calling inplace cuda sampling methods will alloc an RNG state. After del the tensor, the state still remains in GPU memory, causing equal checks at line 538 to fail.